src/components/ConfirmationModal/ConfirmationModal.js

```diff
@@ -0,0 +1,55 @@
+import { useEscape } from '../../hooks/useEscape';
+import './ConfirmationModal.css';
+
+function ConfirmationModal({
+  isActive,
+  closeModal,
+  buttonText,
+  title,
+  handleButton,
+}) {
+  useEscape(handleCloseModal);
+
+  const handleClickOutsideClose = (evt) => {
+    if (evt.target.classList.contains('modal')) {
+      handleCloseModal();
+    }
+  };
+
+  if (isActive) {
+    setTimeout(() => {
+      document.querySelector('.modal').classList.add('modal_opened');
```

Comment:
`document.querySelector` is not allowed to use in `React` because it’s a **very slow** operation for the browser. React was specially created not to search anything in the DOM. So, you can directly operate with the markup inside the components. Or you can use hook `useRef` for the reference.


 You should use the same logics of closing like you do with any other modal in the project. You have `closeModal` for closing any modal. 


```js

  if (isActive) {
    setTimeout(() => {
      closeModal();
    }, 10);
  }
```